### PR TITLE
Implement GET and DELETE methods to API workflow

### PR DESCRIPTION
**Pull request by mfturra**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -54,15 +67,35 @@ def videogame_POST():
         else:
             return 'Request not being understood. Check previous if and elif statements'
 
-    # Successful POST: curl -X POST http://127.0.0.1:5000/videogames -H "Content-Type: application/json" -d '{"title":"Spiderman 2", "platform":"PlayStation"}'
-    # Inaccurate JSON file: curl -X POST http://127.0.0.1:5000/videogames -H "Content-Type: application/json" -d '{"name":"Spiderman 2", "platform":"PlayStation 2"}'
 
+@app.route('/videogames/<uuid>', methods=['GET'])
```
Comment (fahlmant) [inline app.py:71]: Still need GET for `/videogames`

File: app.py [hunk 2]
```diff
@@ -54,15 +67,35 @@ def videogame_POST():
...
+# GET Request Example: curl -X GET http://127.0.0.1:5000/videogames/88cb6cae-f55b-4f65-839c-ab58a5d88e91
+# Query for UUID entry in the Videogame Database
+def info_query(uuid):
+    entry_info = db.session.query(Videogame).filter(Videogame.videogame_id == uuid).first()
+    if entry_info:
+        # print(f"The requested ID is: {entry_info.videogame_id}\nThe title is: {entry_info.videogame_title}\nThe platform is: {entry_info.videogame_platform}")
+        return f"The requested ID is: {entry_info.videogame_id}\nThe title is: {entry_info.videogame_title}\nThe platform is: {entry_info.videogame_platform}\nThe release date is: {entry_info.videogame_releasedate}\nThe publisher is: {entry_info.videogame_publisher}"
```
Comment (fahlmant) [inline app.py:78]: This should return JSON, just like POST does

File: app.py [hunk 3]
```diff
@@ -54,15 +67,35 @@ def videogame_POST():
...
+# DELETE Request Example: curl -X DELETE http://127.0.0.1:5000/videogames/88cb6cae-f55b-4f65-839c-ab58a5d88e91
+# Query for UUID entry and delete entry
+def delete_entry(uuid):
+    entry_info = db.session.query(Videogame).filter(Videogame.videogame_id == uuid).first()
+    if entry_info:
+        db.session.delete(entry_info)
+        db.session.commit()
+        return "Entry has been deleted."
+    else:
+        return "Entry not found."
```
Comment (fahlmant) [inline app.py:93]: These should both return JSON as well

**Review by fahlmant — CHANGES_REQUESTED** [review]
Please update the title and description of the PR to reflect that you're implementing GET, PATCH, and DELETE

File: app.py [hunk 4]
```diff
@@ -1,27 +1,27 @@
...
-        json_dict = json.loads(videogame_entry)
-        return cls(**json_dict)
-
-
-@app.route('/videogames')
-def videogame_GET():
-    return "Main page for video game directory."
+@app.route('/videogames', methods=['GET'])
+def videogame_main():
+    return "Welcome to the main videogames archive page!"
```
Comment (fahlmant) [inline app.py:18]: This should return all videogames in the DB

File: app.py [hunk 5]
```diff
@@ -33,36 +33,65 @@ def videogame_POST():
...
-    assert resource == request.view_args['uuid']
-    return "Pull specific info on video games in database."
+@app.route('/videogames/<uuid>', methods=['GET'])
+# GET Request Example: curl -X GET http://127.0.0.1:5000/videogames/88cb6cae-f55b-4f65-839c-ab58a5d88e91
+# Query for UUID entry in the Videogame Database
+def info_query(uuid):
+    entry_info = db.session.query(Videogame).filter(Videogame.videogame_id == uuid).first()
+    if entry_info:
+        # print(f"The requested ID is: {entry_info.videogame_id}\nThe title is: {entry_info.videogame_title}\nThe platform is: {entry_info.videogame_platform}")
+        videogame_query = "The requested ID is: {entry_info.videogame_id}\nThe title is: {entry_info.videogame_title}\nThe platform is: {entry_info.videogame_platform}\nThe release date is: {entry_info.videogame_releasedate}\nThe publisher is: {entry_info.videogame_publisher}"
```
Comment (fahlmant) [inline app.py:75]: You don't need all the other data. This should look like the information that comes back from POST, just the json body with the video game info.

File: app.py [hunk 6]
```diff
@@ -33,36 +33,65 @@ def videogame_POST():
...
+
+@app.route('/videogames/<uuid>', methods=['DELETE'])
+# DELETE Request Example: curl -X DELETE http://127.0.0.1:5000/videogames/88cb6cae-f55b-4f65-839c-ab58a5d88e91
+# Query for UUID entry and delete entry
+def delete_entry(uuid):
+    entry_info = db.session.query(Videogame).filter(Videogame.videogame_id == uuid).first()
+    if entry_info:
+        db.session.delete(entry_info)
+        db.session.commit()
+        query_deleted = "Entry has been deleted."
```
Comment (fahlmant) [inline app.py:90]: What does this look like when it's jsonified?

Comment (mfturra) [reply app.py:90]: Updated the script to produce a json message when handling different outcomes.

File: app.py [hunk 7]
```diff
@@ -1,29 +1,29 @@
...
-        return cls(**json_dict)
-
-
-@app.route('/videogames')
-def videogame_GET():
-    return "Main page for video game directory."
+@app.route('/videogames', methods=['GET'])
+def videogame_main():
+    query_all = db.session.query(Videogame).all()
+    return f"These are the videogame database entries: {query_all}"
```
Comment (fahlmant) [inline app.py:19]: JSON :)

File: app.py [hunk 8]
```diff
@@ -16,7 +16,27 @@
...
+            "releasedate": entry.videogame_releasedate,
+            "publisher": entry.videogame_publisher
+        }
+        
+        # Combine all entries into the all_entries list
+        all_entries.append(entry_info)
+
+    response = {
+        "message": "These are all the videogame database entries",
+        "entries": all_entries
```
Comment (fahlmant) [inline app.py:37]:
```suggestion
        "items": all_entries
```

**Review by fahlmant — APPROVED** [review]
LGTM

**Pull request merged**